UNITTESTS/stubs/hal/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(mbed-stubs-hal)
+
+target_compile_definitions(mbed-stubs-hal
+    PRIVATE
+        DEVICE_PWMOUT
+        DEVICE_WATCHDOG
+        MBED_WDOG_ASSERT=1
+)
+
+target_sources(mbed-stubs-hal
+    PRIVATE
```

Comment:
sources should always be PRIVATE, otherwise you're compiling them in the app that link against the lib